src/main/java/io/confluent/idesidecar/restapi/application/ProxyProcessorBeanProducers.java

```diff
@@ -37,13 +61,29 @@ public Processor<ClusterProxyContext, Future<ClusterProxyContext>> clusterProxyP
       ClusterProxyRequestProcessor clusterProxyRequestProcessor
   ) {
     return Processor.chain(
-        new ConnectionProcessor<>(connectionStateManager),
+        connectionProcessorClusterProxyContext,
         clusterAuthenticationProcessor,
         clusterInfoProcessor,
         clusterStrategyProcessor,
         clusterProxyProcessor,
         clusterProxyRequestProcessor,
-        new EmptyProcessor<>()
+        emptyProcessorClusterProxyContext
+    );
+
+  }
+
+  @Produces
+  @Singleton
+  @Named("RBACProxyProcessor")
+  public Processor<ProxyContext, Future<ProxyContext>> RbacProxyProcessor(
+      RBACProxyProcessor rbacProxyProcessor
+  ) {
+    return Processor.chain(
+        new ConnectionProcessor<>(connectionStateManager),
+        controlPlaneAuthenticationProcessor,
+        rbacProxyProcessor,
+        new ProxyRequestProcessor(webClientFactory, vertx),
+        emptyProcessorProxyContext
```

Comment:
I think the ordering here needs an update: the `rbacProxyProcessor` sets the proxy request parameters, which should then be modified by the `controlPlaneAuthenticationProcessor` (instead of other way round).